Plugin/Model/ResourceModel/Category/TreePlugin.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Snowdog\Menu\Plugin\Model\ResourceModel\Category;
+
+use Magento\Catalog\Model\ResourceModel\Category\Tree;
+use Magento\Framework\App\RequestInterface;
+
+class TreePlugin
+{
+    private RequestInterface $request;
+
+    public function __construct(
+        RequestInterface $request
+    ) {
+        $this->request = $request;
+    }
+
+    public function beforeAddCollectionData(Tree $subject, $collection = null): array
+    {
+        $postData = $this->request->getPost();
+        $storeId = $postData['store_id'];
+
+        $collection->setProductStoreId(
+            $storeId
+        )->setStoreId(
+            $storeId
+        );
```

Comment:
Maybe we should skip `setProductStoreId` and `setStoreId` if the post param is missing?